app/src/Talk/TalkCommentEntity.php

```diff
@@ -1,36 +1,14 @@
 <?php
 namespace Talk;
 
-class TalkCommentEntity
-{
-    private $data;
-
-    /**
-     * Create new TalkCommentEntity
-     *
-     * @param Object $data Model data retrieved from API
-     */
-    public function __construct($data)
-    {
-        $this->data = $data;
-    }
-
-    public function getRating()
-    {
-        if (!isset($this->data->rating)) {
-            return null;
-        }
-
-        return $this->data->rating;
-    }
+use Application\BaseCommentEntity;
 
-    public function getUserDisplayName()
+class TalkCommentEntity extends BaseCommentEntity
+{
+    public function __construct(\stdClass $data)
     {
-        if (!isset($this->data->user_display_name)) {
-            return null;
-        }
-
-        return $this->data->user_display_name;
+        parent::__construct($data);
+        $this->commentUri = isset($this->data->uri) ? $this->data->uri : null;
```

Comment:
This checks for `$this->data->uri` and the other checks for `$data->comment_uri` (soon to be `$this->data->comment_uri`). This is a great example for why extending the base class makes for more readable code!